code/game/objects/items/devices/scanners/plant_analyzer.dm

```diff
@@ -0,0 +1,275 @@
+		playsound(src, SFX_INDUSTRIAL_SCAN, 20, TRUE, -2, TRUE, FALSE)
+		var/mob/living/L = target
+		if(L.mob_biotypes & MOB_PLANT)
+			plant_biotype_health_scan(target, user)
+			return TRUE
+
+	return analyze(user, target)
+
+/// Same as above, but with right click. Right-clicking scans for chemicals.
+/obj/item/plant_analyzer/pre_attack_secondary(atom/target, mob/living/user)
+	if(user.combat_mode || !user.can_read(src))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+
+	return do_plant_chem_scan(target, user) ? SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN : SECONDARY_ATTACK_CONTINUE_CHAIN
+
+/*
+ * Scan the target on chemical scan mode. This prints chemical genes and reagents to the user.
+ *
+ * scan_target - the atom we're scanning
+ * user - the user doing the scanning.
+ *
+ * returns FALSE if it's not an object or item that does something when we scan it.
+ * returns TRUE if we can scan the object, and outputs the message to the USER.
+ */
+/obj/item/plant_analyzer/proc/do_plant_chem_scan(atom/scan_target, mob/user)
+	if(isliving(scan_target))
+		var/mob/living/L = scan_target
+		if(L.mob_biotypes & MOB_PLANT)
+			plant_biotype_chem_scan(scan_target, user)
+			return TRUE
+
+	return analyze(user, scan_target)
+
+/*
+ * Scan a living mob's (with MOB_PLANT biotype) health with the plant analyzer. No wound scanning, though.
+ *
+ * scanned_mob - the living mob being scanned
+ * user - the person doing the scanning
+ */
+/obj/item/plant_analyzer/proc/plant_biotype_health_scan(mob/living/scanned_mob, mob/living/carbon/human/user)
+	user.visible_message(
+		span_notice("[user] analyzes [scanned_mob]'s vitals."),
+		span_notice("You analyze [scanned_mob]'s vitals.")
+		)
+
+	healthscan(user, scanned_mob, advanced = TRUE)
+	add_fingerprint(user)
+
+/*
+ * Scan a living mob's (with MOB_PLANT biotype) chemical contents with the plant analyzer.
+ *
+ * scanned_mob - the living mob being scanned
+ * user - the person doing the scanning
+ */
+/obj/item/plant_analyzer/proc/plant_biotype_chem_scan(mob/living/scanned_mob, mob/living/carbon/human/user)
+	user.visible_message(
+		span_notice("[user] analyzes [scanned_mob]'s bloodstream."),
+		span_notice("You analyze [scanned_mob]'s bloodstream.")
+		)
+	chemscan(user, scanned_mob)
+	add_fingerprint(user)
+
+/obj/item/plant_analyzer/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "PlantAnalyzer", "Plant Analyzer")
+		ui.open()
+
+/obj/item/plant_analyzer/ui_data(mob/user)
+	return last_scan_data
+
+/obj/item/plant_analyzer/ui_static_data(mob/user)
+	var/list/data = list()
+	data["cycle_seconds"] = HYDROTRAY_CYCLE_DELAY / 10
+	data["trait_db"] = list()
+	for(var/trait_path in subtypesof(/datum/plant_gene/trait))
+		var/datum/plant_gene/trait/trait = new trait_path
+		var/trait_data = list(list(
+			"path" = trait.type,
+			"name" = trait.name,
+			"icon" = trait.icon,
+			"description" = trait.description
+		))
+		data["trait_db"] += trait_data
```

Comment:
This should also be cached, we can do this once - stick it in a static list - and re-use that.
`ui_static_data` is called every time the UI is opened after all.